### PR TITLE
Feature/additional attendees list

### PR DESCRIPTION
**Pull request by olufemi-olumaiyegun**
Update the mongodb participant collection to keep track of how long a user is deafened or absent

### CODE REVIEW

File: src/app/events/tracking/HandleParticipantDuringEvent.ts [hunk 1]
```diff
@@ -47,7 +68,43 @@ const HandleParticipantDuringEvent = async (oldState: VoiceState, newState: Voic
...
+			});
+			const participant = await retrieveActiveParticipant({ id: oldState.id, tag: oldState.member?.user.tag }, oldState.channelId, `${oldState.member?.user.tag}`);
+			if (!leavingUserQuery) {
+				const minutesAttended = dayjs().diff(participant?.startTime, 'm', true);
+				const resultInsert: InsertOneWriteOpResult<POAPParticipant> | void = await addtionalPoapParticipantsCol.insertOne({
+					discordUserId: oldState.id,
+					discordUserTag: oldState.member?.user.tag,
+					voiceChannelId: oldState.channelId,
+					minutesAttended:  minutesAttended,
+					timeLeft: dayjs().toISOString(),
```
Comment (amaredeus) [inline src/app/events/tracking/HandleParticipantDuringEvent.ts:85]: is timeLeft the current date in ISO?

File: src/app/events/tracking/HandleParticipantDuringEvent.ts [hunk 2]
```diff
@@ -96,14 +153,45 @@ const isChannelActivePOAPEvent = async (
...
-const removeDeafenedUser = async (channelId: string | null, guildId: string | null, userId: string | null) => {
+const removeDeafenedUser = async (channelId: string | null, guildId: string | null, userId: string | null, userTag: string | undefined) => {
 	if (channelId == null || guildId == null || userId == null) {
 		return;
 	}
 	
 	const db: Db = await MongoDbUtils.connect(constants.DB_NAME_DEGEN);
 	const poapParticipantsCol: Collection<POAPParticipant> = db.collection(constants.DB_COLLECTION_POAP_PARTICIPANTS);
-	
+	const addtionalPoapParticipantsCol: Collection<POAPParticipant> = db.collection(constants.DB_COLLECTION_ADDITIONAL_POAP_PARTICIPANTS_LIST);
```
Comment (amaredeus) [inline src/app/events/tracking/HandleParticipantDuringEvent.ts:163]: These are a lot of db operations for a single user when an event change occurs. Keeping in mind that there may be hundreds of users changing their status, can we condense some of these operations?

**Review by amaredeus — CHANGES_REQUESTED** [review]

File: src/app/events/tracking/HandleParticipantDuringEvent.ts (hunk 1 above)
Comment (olufemi-olumaiyegun) [reply src/app/events/tracking/HandleParticipantDuringEvent.ts]: yes

Comment (olufemi-olumaiyegun) [reply src/app/events/tracking/HandleParticipantDuringEvent.ts]: it saves the time a user leaves the voice channel

**Review by amaredeus — DISMISSED** [review]

**Review by amaredeus — APPROVED** [review]

**Pull request merged**